src/core/execute.c

```diff
@@ -201,6 +202,63 @@ static const char *exec_context_tty_path(const ExecContext *context) {
         return "/dev/console";
 }
 
+static int exec_context_tty_size(const ExecContext *context, unsigned *ret_rows, unsigned *ret_cols) {
+        unsigned rows = UINT_MAX, cols = UINT_MAX;
+        const char *tty;
+        int r;
+
+        assert(context);
+        assert(ret_rows);
+        assert(ret_cols);
+
+        tty = skip_dev_prefix(exec_context_tty_path(context));
+
+        if (context->tty_rows == UINT_MAX) {
+                _cleanup_free_ char *key = NULL, *value = NULL;
+
+                key = strjoin("systemd.tty.", tty, ".rows");
+                if (!key)
+                        return -ENOMEM;
+
+                r = proc_cmdline_get_key(key, 0, &value);
+                if (r < 0)
+                        log_debug_errno(r, "Failed to read %s from kernel cmdline, ignoring: %m", key);
+                if (r > 0) {
+                        r = safe_atou(value, &rows);
+                        if (r == -EINVAL)
+                                log_debug_errno(r, "Failed to parse %s=%s, ignoring", key, value);
+                        else if (r < 0)
+                                return r;
```

Comment:
Hopefully, `safe_atou()` always return `-EINVAL` on failure. But for safety, I think we should always ignore the failure. Otherwise, potentially, many services may fail to start when an invalid kernel command line option is specified.